src/stan_math_backend/Transform_Mir.ml

```diff
@@ -295,15 +295,15 @@ let read_constrain_dims constrain_transform st =
       constrain_get_dims st
   | _ -> SizedType.get_dims st
 
-let data_serializer_read loc out_constrained_st =
+let plain_deserializer_read loc out_constrained_st =
   let ut = SizedType.to_unsized out_constrained_st in
-  let dims = SizedType.get_dims_io out_constrained_st in
+  let dims = SizedType.get_dims out_constrained_st in
   let emeta = Expr.Typed.Meta.create ~loc ~type_:ut ~adlevel:AutoDiffable () in
   Expr.(
     Helpers.(
-      internal_funapp FnReadDataSerializer dims Typed.Meta.{emeta with type_= ut}))
+      internal_funapp FnReadSerializer dims Typed.Meta.{emeta with type_= ut}))
```

Comment:
Stupid question: why does a function called `deserializer_read` create a call to `ReadSerializer`?

Reply:
Not at all a stupid question, though the answer might be. It's probably for the same reason the code gen for this function uses a OCaml variable called `serializer_in` which corresponds to the C++ declaration `stan::io::deserializer in__`...
We haven't been the tidiest with our naming conventions. I've been trying to fix them as I noticed, but these are ones I missed and will fix up now